exchanges/poloniex/poloniex_futures.go

```diff
@@ -0,0 +1,741 @@
+	}
+	if marginMode == "" {
+		return nil, margin.ErrInvalidMarginType
+	}
+	if positionSide == "" {
+		return nil, order.ErrSideIsInvalid
+	}
+	if leverage <= 0 {
+		return nil, order.ErrSubmitLeverageNotSupported
+	}
+	var resp *FuturesLeverage
+	return resp, e.SendAuthenticatedHTTPRequest(ctx, exchange.RestSpot, request.UnAuth, http.MethodPost, positionPathV3+"leverage", nil, &map[string]string{
+		"symbol":  symbol,
+		"mgnMode": marginMode,
+		"posSide": positionSide,
+		"lever":   strconv.FormatInt(leverage, 10),
+	}, &resp)
+}
+
+// SwitchPositionMode switch the current position mode. Please ensure you do not have open positions and open orders under this mode before the switch.
+// Position mode, HEDGE: LONG/SHORT, ONE_WAY: BOTH
+func (e *Exchange) SwitchPositionMode(ctx context.Context, positionMode string) error {
+	if positionMode == "" {
+		return errPositionModeInvalid
+	}
+	return e.SendAuthenticatedHTTPRequest(ctx, exchange.RestSpot, request.UnAuth, http.MethodPost, positionPathV3+"mode", nil, map[string]string{"posMode": positionMode}, &struct{}{})
+}
+
+// GetPositionMode get the current position mode.
+func (e *Exchange) GetPositionMode(ctx context.Context) (string, error) {
+	resp := &struct {
+		PositionMode string `json:"posMode"`
+	}{}
+	return resp.PositionMode, e.SendAuthenticatedHTTPRequest(ctx, exchange.RestSpot, request.UnAuth, http.MethodGet, positionPathV3+"mode", nil, nil, &resp)
+}
+
+// GetFuturesOrderBook get market depth data of the designated trading pair
+func (e *Exchange) GetFuturesOrderBook(ctx context.Context, symbol string, depth, limit uint64) (*FuturesOrderbook, error) {
+	if symbol == "" {
+		return nil, currency.ErrSymbolStringEmpty
+	}
+	params := url.Values{}
+	params.Set("symbol", symbol)
+	if depth > 0 {
+		params.Set("scale", strconv.FormatUint(depth, 10))
+	}
+	if limit > 0 {
+		params.Set("limit", strconv.FormatUint(limit, 10))
+	}
+	var resp *FuturesOrderbook
+	return resp, e.SendHTTPRequest(ctx, exchange.RestSpot, request.UnAuth, common.EncodeURLValues(marketsPathV3+"orderBook", params), &resp)
+}
+
+// GetFuturesKlineData retrieves K-line data of the designated trading pair
+func (e *Exchange) GetFuturesKlineData(ctx context.Context, symbol string, interval kline.Interval, startTime, endTime time.Time, limit uint64) ([]*FuturesCandle, error) {
+	if symbol == "" {
+		return nil, currency.ErrSymbolStringEmpty
+	}
+	params := url.Values{}
+	intervalString, err := intervalToString(interval)
+	if err != nil {
+		return nil, err
+	}
+	params.Set("symbol", symbol)
+	params.Set("interval", intervalString)
+	if !startTime.IsZero() && !endTime.IsZero() {
+		if err := common.StartEndTimeCheck(startTime, endTime); err != nil {
+			return nil, err
+		}
+		params.Set("sTime", strconv.FormatInt(startTime.UnixMilli(), 10))
+		params.Set("eTime", strconv.FormatInt(endTime.UnixMilli(), 10))
+	}
+	if limit > 0 {
+		params.Set("limit", strconv.FormatUint(limit, 10))
+	}
+	var resp FuturesCandles
```

Comment:
```suggestion
	var resp []*FuturesCandle
```

and RM the type below as it is not needed